pkgs/development/r-modules/generic-builder.nix

```diff
@@ -1,10 +1,17 @@
-{ stdenv, R, xvfb_run, utillinux }:
+{ stdenv, R, tzdata, gettext, gfortran, libcxx, xvfb_run, utillinux
+
+, Cocoa, Foundation
+}:
 
 { name, buildInputs ? [], ... } @ attrs:
 
 stdenv.mkDerivation ({
-  buildInputs = buildInputs ++ [R] ++
-                stdenv.lib.optionals attrs.requireX [utillinux xvfb_run];
+  buildInputs = buildInputs ++ [ R tzdata gettext gfortran ] ++
+                stdenv.lib.optionals attrs.requireX [utillinux xvfb_run]
+    ++ stdenv.lib.optionals stdenv.isDarwin [ Cocoa Foundation ];
+
+  NIX_CFLAGS_COMPILE =
+    stdenv.lib.optionalString stdenv.isDarwin "-I${libcxx}/include/c++/v1";
```

Comment:
It would be much better if R knew about this flag itself. This hack fixes compilation within Nix, but it won't work for people who use R's facilities for compiling / installing packages from within R.